pkg/api/pod/util.go

```diff
@@ -895,6 +902,9 @@ func dropDisabledPodStatusFields(podStatus, oldPodStatus *api.PodStatus, podSpec
 		dropResourcesField(podStatus.InitContainerStatuses)
 		dropResourcesField(podStatus.EphemeralContainerStatuses)
 
+		// Drop pod-level Resources fields from PodStatus
+		podStatus.Resources = nil
```

Comment:
nit: you don't need to drop the resources here. With FG dependencies, this is redundant with the previous block.